plugins/notifications-backend/src/database/DatabaseNotificationsStore.ts

```diff
@@ -194,6 +235,31 @@ export class DatabaseNotificationsStore implements NotificationsStore {
       );
     }
 
+    if (options.metadata) {
+      const count = Object.keys(options.metadata ?? {}).length;
+
+      const notificationMetadataQuery = (tableName: string) => {
+        const mQuery = this.db(tableName).select('notification_id');
+
+        Object.keys(options.metadata ?? {}).forEach(key => {
+          const value = options.metadata?.[key];
+          mQuery.orWhereRaw(`(name_lower LIKE ? AND value_lower LIKE ?)`, [
+            key,
+            `%${value?.toString().toLowerCase()}%`,
+          ]);
+        });
+        mQuery
+          .groupBy('notification_id')
+          .having(this.db.raw(`COUNT(DISTINCT name) = ${count}`));
```

Comment:
i am not sure i understand this one

Reply:
Oh yeah, it's a tricky one.

It's when you want to find a notification where multiple metadata's satisfies several conditions.

For example, you want to find a notification which has all these attributes and they satisfy these conditions:

status = 'pending'
team = 'group:default/mybestteam'
requestId = 'id-3'

It's also discussed for example here:  https://stackoverflow.com/questions/20845975/select-values-from-table-with-multiple-attributes